src/components/CustomSection.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+import Button from "./Button";
+import Grid from "./Grid";
+import Section from "./Section";
+
+const ButtonWrapper = styled.div`
+  margin-top: ${(props) => props.theme.spacing(4)};
+`;
+
+const CustomSection = ({
+  button,
+  color,
+  components,
+  description,
+  linkComponent,
+  title,
+}) => {
+  const Link = linkComponent;
+  const renderButton =
+    button && button.label && button.href ? (
+      <Button as={Link} href={button.href} color={button.color || "light"}>
+        {button.label}
+      </Button>
+    ) : null;
+
+  return (
+    <Section color={color}>
+      <Section.Title>{title}</Section.Title>
+      <Grid>
+        <Section.Text>{description}</Section.Text>
+        {components}
+        <ButtonWrapper>{renderButton}</ButtonWrapper>
```

Comment:
Since the custom section block always comes immediately before the getting started section, I don't think the button at the end of this section will be very commonly used. Maybe skip it here to simplify this component and corresponding config.